components/HackathonModal.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect } from 'react';
+import Modal from 'react-modal';
+import Image from '../components/Image';
+import Link from 'next/link';
+import styles from '../styles/components/HackathonModal.module.scss';
+import { website_config } from '../config';
+
+const MODAL_DISPLAY_INTERVAL = 2 * 24 * 60 * 60 * 1000; // 2 days in milliseconds
+
+interface HackathonModalProps {
+  onRequestClose: () => void;
+}
+
+const HackathonModal: React.FC<HackathonModalProps> = ({ onRequestClose }) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [assetsLoaded, setAssetsLoaded] = useState<boolean>(false);
+
+  useEffect(() => {
+    const fontFamilies = ['Orbitron', 'Poppins', 'Motorblock'];
+
+    const imageUrls = [
+      '/Hackathon/little-guy.png',
+      '/Hackathon/top-left.svg',
+      '/Hackathon/top-right.svg',
+      '/Hackathon/bottom-left.svg',
+    ];
+
+    const loadFonts = async () => {
+      try {
+        await Promise.all(
+          fontFamilies.map((family) => document.fonts.load(`1em "${family}"`))
```

Comment:
Might it be worth using the next.js font stuff here? (I get what your trying todo here but realistically these fonts are small, i dont think lazy loading helps here it just makes things more complex), we could load the fonts inline here using next.js and use a dynamic import to dynamically import this module maybe). 


https://nextjs.org/docs/pages/building-your-application/optimizing/fonts